interop/test/ping.spec.ts

```diff
@@ -0,0 +1,164 @@
+  }
+  return await res.json()
+}
+
+describe('ping test', () => {
+  // eslint-disable-next-line complexity
+  it('should ping', async () => {
+    const TRANSPORT = process.env.transport
+    const SECURE_CHANNEL = process.env.security
+    const MUXER = process.env.muxer
+    const isDialer = process.env.is_dialer === 'true'
+    const IP = process.env.ip ?? '0.0.0.0'
+    const timeoutSecs: string = process.env.test_timeout_secs ?? '180'
+
+    const options: Libp2pOptions<{ ping: PingService }> = {
+      start: true,
+      connectionGater: {
+        denyDialMultiaddr: async () => false
+      },
+      services: {
+        ping: pingService()
+      }
+    }
+
+    switch (TRANSPORT) {
+      case 'tcp':
+        options.transports = [tcp()]
+        options.addresses = {
+          listen: isDialer ? [] : [`/ip4/${IP}/tcp/0`]
+        }
+        break
+      case 'webtransport':
+        options.transports = [webTransport()]
+        if (!isDialer) {
+          throw new Error('WebTransport is not supported as a listener')
+        }
+        break
+      case 'webrtc':
+        options.transports = [webRTC()]
+        options.addresses = {
+          listen: isDialer ? [] : [`/ip4/${IP}/udp/0/webrtc`]
+        }
+        break
+      case 'ws':
+        options.transports = [webSockets()]
+        options.addresses = {
+          listen: isDialer ? [] : [`/ip4/${IP}/tcp/0/ws`]
+        }
+        break
+      case 'wss':
+        process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0'
+        options.transports = [webSockets()]
+        options.addresses = {
+          listen: isDialer ? [] : [`/ip4/${IP}/tcp/0/wss`]
+        }
+        break
+      default:
+        throw new Error(`Unknown transport: ${TRANSPORT ?? '???'}`)
+    }
+
+    let skipSecureChannel = false
+    let skipMuxer = false
+    switch (TRANSPORT) {
+      case 'webtransport':
+      case 'webrtc':
+        skipSecureChannel = true
+        skipMuxer = true
+        break
+      default:
+        // Do nothing
+    }
+
+    if (!skipSecureChannel) {
+      switch (SECURE_CHANNEL) {
+        case 'noise':
+          options.connectionEncryption = [noise()]
+          break
+        case 'quic':
+          options.connectionEncryption = [noise()]
+          break
+        default:
+          throw new Error(`Unknown secure channel: ${SECURE_CHANNEL ?? ''}`)
+      }
+    } else {
+      // Libp2p requires at least one encryption module. Even if unused.
+      options.connectionEncryption = [noise()]
+    }
+
+    if (!skipMuxer) {
+      switch (MUXER) {
+        case 'mplex':
+          options.streamMuxers = [mplex()]
+          break
+        case 'yamux':
+          options.streamMuxers = [yamux()]
+          break
+        case 'quic':
+          break
+        default:
+          throw new Error(`Unknown muxer: ${MUXER ?? '???'}`)
+      }
+    }
+
+    const node = await createLibp2p(options)
+
+    try {
+      if (isDialer) {
```

Comment:
Conditional logic in tests is always raises an eyebrow as you can end up not testing what you think you are testing.

Can this be split into two separate tests?

For example the dialer test could skip itself if the node isn't the dialer on this run.

Reply:
Why? Every other [implementation](https://github.com/libp2p/go-libp2p/blob/master/test-plans/cmd/ping/main.go#L161) [does](https://github.com/libp2p/rust-libp2p/blob/master/interop-tests/src/bin/ping.rs#L127) [it](https://github.com/libp2p/test-plans/blob/master/multidim-interop/impl/nim/mainv1.nim#L70) [this](https://github.com/MarcoPolo/zig-libp2p/blob/main/interop/main.zig#L118) way.

Reply:
Looking at the other implementations, none of them are written using a test framework so I don't think they are an example to follow here.

Reply:
Maybe we shouldn’t use the test framework? I only used it because it was convenient. Happy to not use it.

Reply:
Do you feel strongly enough against the test framework as to request changes before merging?

I don’t think we should deviate from what the four other implementations do just follow the convention on this test framework, so I would suggest keeping the conditional as is, and , if requested, moving off the test framework. 

I would be against splitting this into two tests.

Reply:
I think using a test framework to write tests is the right thing to do, as is integrating with the existing tooling of a project because it makes the code instantly familiar to other developers so I think this is the right decision.

I think regardless of framework tests have a certain structure (setup, exercise, verify, teardown), duplication should be minimised and that conditional logic in tests means that later down the line when something goes wrong the author will have to think about which branch is being executed which is a code smell and should be avoided.

I also think that the first test is the most important since people will use it as a starting point for subsequent tests and am somewhat confused as to why you are pushing back on this so hard.